core/domain/classifier_services.py

```diff
@@ -359,62 +419,52 @@ def delete_classifier_training_job(job_id):
-                                          classifier_id):
-    """Creates an entry for Classifier Exploration Mapping in datastore.
+    training_job_exploration_mapping_models = (
+        classifier_models.TrainingJobExplorationMappingModel.get_models(
+            exp_id, exp_version, state_names))
+    if training_job_exploration_mapping_models is None:
+        return None
+    job_ids = []
+    for mapping_model in training_job_exploration_mapping_models:
+        if mapping_model is None:
+            continue
+        job_ids.append(mapping_model.job_id)
+    classifier_training_job_models = (
+        classifier_models.ClassifierTrainingJobModel.get_multi(job_ids))
+    classifier_training_jobs = []
+    for job_model in classifier_training_job_models:
+        classifier_training_jobs.append(get_classifier_training_job_from_model(
+            job_model))
+
+    # Backfill None's to maintain indexes.
+    for index, mapping_model in enumerate(
+            training_job_exploration_mapping_models):
+        if mapping_model is None:
+            classifier_training_jobs.insert(index, None)
+    return classifier_training_jobs
+
+
+def _create_multi_job_exploration_mappings(job_exploration_mappings):
+    """Creates multiple TrainingJobExplorationMappingModel instances in data
+    store.
+    NOTE: The client is responsible for passing in valid mapping domain objects
+    into this method.
 
     Args:
-        exp_id: str. ID of the exploration.
-        exp_version: int. The exploration version.
-        state_name: str. The state to which the classifier belongs.
-        classifier_id: str. ID of the classifier.
-
-    Returns:
-        str. ID of the classifier exploration mapping instance.
-
-    Raises:
-        Exception: The Classifier-Exploration mapping with id already exists.
+        job_exploration_mappings: list(TrainingJobExplorationMapping). The list
+            of TrainingJobExplorationMapping Domain objects.
     """
-    classifier_exploration_mapping_model = (
-        classifier_models.ClassifierExplorationMappingModel.get_model(
-            exp_id, exp_version, state_name))
-    if classifier_exploration_mapping_model is not None:
-        raise Exception('The Classifier-Exploration mapping with id %s.%s.%s '
-                        'already exists.' % (exp_id, exp_version,
-                                             state_name.encode('utf-8')))
-    # Verify that the corresponding classifier exists.
-    get_classifier_by_id(classifier_id)
-
-    classifier_exploration_mapping = (
-        classifier_domain.ClassifierExplorationMapping(
-            exp_id, exp_version, state_name, classifier_id))
-    classifier_exploration_mapping.validate()
-    mapping_id = classifier_models.ClassifierExplorationMappingModel.create(
-        exp_id, exp_version, state_name, classifier_id)
-    return mapping_id
+    classifier_models.TrainingJobExplorationMappingModel.create_multi(
```

Comment:
OK, well -- if that's all this function does, I think you might as well just inline it where it's used :P